packages/connect-node/src/grpc-protocol.ts

```diff
@@ -65,20 +67,35 @@ export function createGrpcProtocol(
       switch (spec.kind) {
         case MethodKind.Unary:
           return async (req, res) => {
+            const requestHeader = nodeHeaderToWebHeader(req.headers);
             const type = grpcParseContentType(
               req.headers["content-type"] ?? null
             );
+            const timeout = parseGrpcTimeoutFromHeader(
+              requestHeader.get(grpcTimeoutHeader) ?? ""
+            );
```

Comment:
If we fail to parse grpc-timeout, you need to call `endWithGrpcWebTrailer()`. 

Please parse the timeout _after_ checking whether the content-type is valid. We should not try to parse the `grpc-timeout` header if we don't recognize the `content-type` header.